src/composeApolloProvider.tsx

```diff
@@ -0,0 +1,72 @@
+/// <reference path="../typings/main.d.ts" />
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+import {
+    Component,
+    PropTypes,
+} from 'react';
+
+import {
+    Store,
+} from 'redux';
+
+
+import ApolloClient from 'apollo-client';
+
+declare interface ProviderProps {
+    store?: Store<any>;
+    client: ApolloClient;
+}
+
+export default function composeApolloProvider<T>(Provider: React.ComponentClass<any>): any {
```

Comment:
Is this function better than simply having `ApolloProvider` always use `Provider` as the implementation? As long as it's a peer dependency, the developer can choose whatever version of the Redux provider they feel like.